create/application.go

```diff
@@ -102,6 +104,20 @@ func (app *applicationCmd) Run(ctx context.Context, client *api.Client) error {
 		SSHPrivateKey: sshPrivateKey,
 	}
 
+	if !app.SkipRepoAccessCheck {
+		// check if the repository is reachable via the git information service
+		gitInfoClient, err := util.NewGitInformationClient(app.GitInformationServiceURL, client.Token)
+		if err != nil {
+			return err
+		}
+		gitInfoClient.SetLogRetryFunc(util.RetryLogFunc(app.Debug))
+		fmt.Println("Testing repository access")
+		if err := util.TestRepositoryAccess(ctx, gitInfoClient, &newApp.Spec.ForProvider.Git, auth); err != nil {
+			return err
+		}
+		fmt.Println("Repository access successfully tested")
```

Comment:
I gave this a spin locally and it would IMO look quite a bit nicer if this would make use of our spinner system. It's quite easy to use:

```golang
msg := message{text: "testing repository access", icon: "🔐"}
spinner, err := format.NewSpinner(msg.progress(), msg.progress())
if err != nil {
	return err
}

spinner.Start()
if err := doWork(); err != nil {
	spinner.StopFail()
	return err
}
spinner.Stop()
```

feel free to choose another icon, I just put something that sort of fits.

Reply:
I added the spinner. Looks better now indeed.

Reply:
Still need to do more DRY there...

Reply:
Ok, as we use exactly the same for `create` and `update`, I moved the logic into a different package.